### ISSUE TITLE
[cata] Fix contrib tools build on Linux

### ISSUE THREAD

**Issue by walker0643** [cmangos/issues#449]
The patch below fixes the cmake build on linux for the contrib tools: ad, MoveMapGen, vmap_assembler, and vmapextractor...
  http://pastebin.ca/2831885

Much of the patch was taken from mangoszero:
  https://github.com/mangoszero/server/commit/3080b49b9a0ec8f942f5331e71fd1fb1e46332f4

Tested on Ubuntu 12.04 64-bit, cata branch.

**Comment by evil-at-wow** [comment]
@stfx Thanks for bringing this to my attention.

Actually, I've already started making changes to the contrib tools to fix the cmake builds (to be able to test a patch by @Schmoozerd afterwards). These tools have always been a huge pain to build on non Windows systems. And it's not just Linux - I'm sure @cala and others would love to see these tools build out of the box on Mac OS X too!

I ran into more issues than the changes here suggest, so I'm fairly sure I won't be able to build the contrib tools with this patch on my Linux system. And then there's the difference between the cata branch and the other branches for the tools that need to extract client data: cata needs stormlib, the other branches are still using libmpq. 

If you don't mind, I'll continue fixing these tools in the other branches first, then after a merge-around I'll give the cata branch a go and see what part of this patch is still needed.
